runelite-client/src/main/java/net/runelite/client/plugins/fps/FpsPlugin.java

```diff
@@ -102,4 +118,59 @@ protected void shutDown() throws Exception
 	{
 		drawManager.unregisterEveryFrameListener(drawListener);
 	}
+
+	@Schedule(
+			asynchronous = true,
+			period = 5,
+			unit = ChronoUnit.SECONDS
+	)
+	public long getPingToCurrentWorld()
+	{
+		InetAddress host;
+		if (client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			try
+			{
+				host = InetAddress.getByName(client.getWorldHostname());
+			}
+			catch (UnknownHostException he)
+			{
+				log.warn("Cannot ping host", he);
+				return -1;
+			}
+
+			Instant start = Instant.now();
+			Socket sock = null;
+			try
+			{
+				sock = new Socket(host, 443);
+			}
+			catch (Exception e)
+			{
+				log.warn("Could not create new socket", e);
+				return -1;
+			}
+			finally
+			{
+				try
```

Comment:
This is really weird.
Is the socket object implementing AutoClosable? https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html

That would let you write a much more concise try catch

Reply:
That's the thing.. This isn't a try-with-resources, this is a normal try catch.

Reply:
Yes, so your first comment is correct and @CustardGod could rewrite it using a try-with-resources statement to have it auto-close.

Reply:
Updated, please take a look